### PR TITLE
fix(site): fix week range for insights

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

**Review by buenos-nachos — APPROVED** [review]
This makes sense to me, and it looks really clean!

File: site/src/pages/TemplatePage/TemplateInsightsPage/utils.ts [hunk 1]
```diff
@@ -1,50 +1,8 @@
...
-function toISOLocal(d: Date) {
-  return format(d, "yyyy-MM-dd'T'HH:mm:ssxxx");
-}
+import { startOfDay, subDays } from "date-fns";
 
 export const lastWeeks = (numberOfWeeks: number) => {
   const now = new Date();
-  const startDate = startOfWeek(subWeeks(now, numberOfWeeks));
-  const endDate = isSunday(now) ? endOfDay(now) : endOfWeek(subWeeks(now, 1));
+  const endDate = startOfDay(subDays(now, 1));
```
Comment (mafredri) [inline site/src/pages/TemplatePage/TemplateInsightsPage/utils.ts:5]: Should this perhaps be just `startOfDay(now)`? Let's say we have `now = 2023-10-10T15:00:00` (Tuesday), `subDays` would put us at `2023-10-09T15:00:00` and `startOfDay` would give us `2023-10-09T00:00:00` meaning we're only looking at data up to and including Sunday.

Comment (BrunoQuaresma) [reply site/src/pages/TemplatePage/TemplateInsightsPage/utils.ts:5]: startOfDay does something like 2023-10-09T23:59:59 but the API only accepts dates as "00:00:00" at the end.

Comment (BrunoQuaresma) [reply site/src/pages/TemplatePage/TemplateInsightsPage/utils.ts:5]: Ahhh sorry, I missread your comment. I will give it a try

**Pull request merged**